dolphinscheduler-common/src/main/java/org/apache/dolphinscheduler/common/utils/S3Utils.java

```diff
@@ -175,21 +175,30 @@ public String getFileName(ResourceType resourceType, String tenantCode, String f
     }
 
     @Override
-    public void download(String tenantCode, String srcFilePath, String dstFile, boolean deleteSource, boolean overwrite) throws IOException {
+    public void download(String tenantCode, String srcFilePath, String dstFilePath, boolean deleteSource, boolean overwrite) throws IOException {
+        File dstFile = new File(dstFilePath);
+        if (dstFile.exists() && !overwrite) {
+            logger.info("the destination path {} already exists, download operation will be ignored", dstFilePath);
+            return;
+        }
+        if (dstFile.isDirectory()) {
+            org.apache.commons.io.FileUtils.deleteDirectory(dstFile);
+        } else {
+            org.apache.commons.io.FileUtils.createParentDirectories(dstFile);
+        }
```

Comment:
Could you keep this consistent with `org.apache.dolphinscheduler.common.utils.HadoopUtils#copyHdfsToLocal`?

Reply:
> Could you keep this consistent with `org.apache.dolphinscheduler.common.utils.HadoopUtils#copyHdfsToLocal`?

The processing logic on both sides is different, and the return value type of the function is also different. I suggest that another PR can be opened for this.

Reply:
> > Could you keep this consistent with `org.apache.dolphinscheduler.common.utils.HadoopUtils#copyHdfsToLocal`?
> 
> The processing logic on both sides is different, and the return value type of the function is also different. I suggest that another PR can be opened for this.

You only need to `overwrite` this part to be consistent. The rest can remain the same.

Reply:
> Could you keep this consistent with `org.apache.dolphinscheduler.common.utils.HadoopUtils#copyHdfsToLocal`?

The copyHdfsToLocal indirectly causes an error when the dstFile path is a directory, but our code deletes it to avoid this

Reply:
> > > Could you keep this consistent with `org.apache.dolphinscheduler.common.utils.HadoopUtils#copyHdfsToLocal`?
> > 
> > 
> > The processing logic on both sides is different, and the return value type of the function is also different. I suggest that another PR can be opened for this.
> 
> You only need to `overwrite` this part to be consistent. The rest can remain the same.


Wait a minute, let me have a look

Reply:
> > > > Could you keep this consistent with `org.apache.dolphinscheduler.common.utils.HadoopUtils#copyHdfsToLocal`?
> > > 
> > > 
> > > The processing logic on both sides is different, and the return value type of the function is also different. I suggest that another PR can be opened for this.
> > 
> > 
> > You only need to `overwrite` this part to be consistent. The rest can remain the same.
> 
> Wait a minute, let me have a look



> > > Could you keep this consistent with `org.apache.dolphinscheduler.common.utils.HadoopUtils#copyHdfsToLocal`?
> > 
> > 
> > The processing logic on both sides is different, and the return value type of the function is also different. I suggest that another PR can be opened for this.
> 
> You only need to `overwrite` this part to be consistent. The rest can remain the same.


The code has been updated